src/Eshopworld.WorkerProcess/Infrastructure/SystemTimer.cs

```diff
@@ -20,22 +20,25 @@ public SystemTimer()
         }
 
         /// <inheritdoc />
-        public async Task ExecutePeriodicallyIn(TimeSpan interval,Func<Task<TimeSpan>> executor)
+        public async Task ExecutePeriodicallyIn(TimeSpan interval,Func<CancellationToken,Task<TimeSpan>> executor)
         {
-            await Task.Delay(interval,_cancellationTokenSource.Token).ContinueWith(async t=>
+            _cancellationTokenSource.Token.ThrowIfCancellationRequested();
```

Comment:
So once we stop the timer, we cannot reuse it?
Is that how it is used in the code?

Reply:
the token should not be resused. But if we want to restart the lease again after being stopped i think we can recreate a new token every time we start the Lease. Should not be a big change

Reply:
ok, so assuming that this is once-off execution

Reply:
moved creation of cancellationtoken within ExecutePeriodicallyIn method, to allow restarting the timer with the workerlease